gems/smithy-client/sig/smithy-client/shapes.rbs

```diff
@@ -2,7 +2,7 @@ module Smithy
   module Client
     module Shapes
       class Shape
-        def initialize: (?Hash[Symbol, untyped]) -> void
+        def initialize: (?Hash[Symbol, untyped]) ?{ (self) -> untyped } -> void
```

Comment:
I think you can make untyped as Shape here since it yields self or does that not work?

Reply:
Tne `untyped` here is the return of the provided block.  The `self` is already what we yield. 

I could change the untyped to `void` to indicate that the return value isn't *used* by our code here.